props/blaster.h

```diff
@@ -93,9 +93,10 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
-  bool auto_firing_ = false;
   int shots_fired_ = 0;
+  bool auto_firing_ = false;
   bool is_jammed_ = false;
+  bool empty_ = false;
```

Comment:
I don't think we need this variable.

I think we want something like this instead:

bool IsEmpty() const {
  return max_shosts_ != -1 && shots_fired_ >= max_shots_;
}

Then instead of checking empty_, we'll just call IsEmpty()